lib/semantics/check-declarations.cc

```diff
@@ -84,7 +91,18 @@ void CheckHelper::Check(Symbol &symbol) {
   auto save{messages_.SetLocation(symbol.name())};
   context_.set_location(symbol.name());
   if (DeclTypeSpec * type{symbol.GetType()}) {
-    Check(*type);
+    bool canHaveAssumedParameter{IsNamedConstant(symbol) ||
+        IsAssumedLengthCharacterFunction(symbol) ||
+        symbol.test(Symbol::Flag::ParentComp)};
+    if (const auto *object{symbol.detailsIf<ObjectEntityDetails>()}) {
+      canHaveAssumedParameter |= object->isDummy() ||
+          (object->isFuncResult() &&
+              type->category() == DeclTypeSpec::Character);
+    } else {
+      canHaveAssumedParameter |=
+          symbol.has<AssocEntityDetails>() || symbol.has<DerivedTypeDetails>();
```

Comment:
The removal of `DerivedTypeDetails` disappeared compared to the original PR.

Reply:
Thanks for noticing.